examples/cfd/xaeronet/surface/train.py

```diff
@@ -0,0 +1,383 @@
+parent_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+sys.path.append(parent_dir)
+
+from dataloader import create_dataloader
+from utils import (
+    find_bin_files,
+    save_checkpoint,
+    load_checkpoint,
+    count_trainable_params,
+)
+
+
+@hydra.main(version_base="1.3", config_path="conf", config_name="config")
+def main(cfg: DictConfig) -> None:
+
+    # Enable cuDNN auto-tuner
+    torch.backends.cudnn.benchmark = cfg.enable_cudnn_benchmark
+
+    # Instantiate the distributed manager
+    DistributedManager.initialize()
+    dist = DistributedManager()
+    device = dist.device
+    print(f"Rank {dist.rank} of {dist.world_size}")
+
+    # Instantiate the writers
+    if dist.rank == 0:
+        writer = SummaryWriter(log_dir="tensorboard")
+        initialize_wandb(
+            project="aws_drivaer",
+            entity="Modulus",
+            name="aws_drivaer",
+            mode="disabled",
+            group="group",
+            save_code=True,
+        )
+
+    # AMP Configs
+    amp_dtype = torch.bfloat16
+    amp_device = "cuda"
+
+    # Find all .bin files in the directory
+    train_dataset = find_bin_files(to_absolute_path(cfg.partitions_path))
+    valid_dataset = find_bin_files(to_absolute_path(cfg.validation_partitions_path))
+
+    # Prepare the stats
+    with open(to_absolute_path(cfg.stats_file), "r") as f:
+        stats = json.load(f)
+    mean = stats["mean"]
+    std = stats["std_dev"]
+
+    # Create DataLoader
+    train_dataloader = create_dataloader(
+        train_dataset,
+        mean,
+        std,
+        batch_size=1,
+        prefetch_factor=None,
+        use_ddp=True,
+        num_workers=4,
+    )
+    # graphs is a list of graphs, each graph is a list of partitions
+    graphs = [graph_partitions for graph_partitions, _ in train_dataloader]
+
+    if dist.rank == 0:
+        validation_dataloader = create_dataloader(
+            valid_dataset,
+            mean,
+            std,
+            batch_size=1,
+            prefetch_factor=None,
+            use_ddp=False,
+            num_workers=4,
+        )
+        validation_graphs = [
+            graph_partitions for graph_partitions, _ in validation_dataloader
+        ]
+        validation_ids = [id[0] for _, id in validation_dataloader]
+        print(f"Training dataset size: {len(graphs)*dist.world_size}")
+        print(f"Validation dataset size: {len(validation_dataloader)}")
+
+    ######################################
+    # Training #
+    ######################################
+
+    # Initialize model
+    model = MeshGraphNet(
+        input_dim_nodes=6,
+        input_dim_edges=4,
+        output_dim=4,
+        processor_size=cfg.num_message_passing_layers,
+        aggregation="sum",
+        hidden_dim_node_encoder=256,
+        hidden_dim_edge_encoder=256,
+        hidden_dim_node_decoder=256,
+        mlp_activation_fn="relu",
+        do_concat_trick=True,
```

Comment:
Use value from config? Same for `num_processor_checkpoint_segments` - there is already a setting in `config.yaml`.

Reply:
Yes I fixed this but forgot to push. Please see the last commit.